src/Confluent.Kafka/Producer.cs

```diff
@@ -302,6 +302,31 @@ public Producer(IEnumerable<KeyValuePair<string, object>> config, bool manualPol
             LibRdKafka.conf_set_log_cb(configPtr, logDelegate);
             LibRdKafka.conf_set_stats_cb(configPtr, statsDelegate);
 
+            // Note: changing the default value of produce.offset.report at the binding level is less than
+            // ideal since it means the librdkafka configuration docs will no longer completely match the 
+            // .NET client. The default should probably be changed in librdkafka as well.
+            
+            if (this.defaultTopicConfig != null)
+            {
+                var topicConfigHandle = SafeTopicConfigHandle.Create();
+                this.defaultTopicConfig.ToList().ForEach(
+                    (kvp) => { topicConfigHandle.Set(kvp.Key, kvp.Value.ToString()); }
+                );
+                if (this.defaultTopicConfig.FirstOrDefault(prop => prop.Key == "produce.offset.report").Value == null)
+                {
+                    topicConfigHandle.Set("produce.offset.report", "true");
+                }
+                LibRdKafka.conf_set_default_topic_conf(configPtr, topicConfigHandle.DangerousGetHandle());
```

Comment:
currently, set_default_topic_conf is not called at all in the producer. Maybe I took this out since these properties can all be set at the top level now... I think I vaguely remember that. however the current code is not good since it just ignores "default.topic.config" without warning. I think it's best to leave the default.topic.config functionality in for now and warn it's depreciated in docs (which are non-existent). Then remove it at some point in the future, throwing an exception if it's used.

Reply:
See earlier comment on not using set_default_topic_conf().